src/core/server/http/http_service.ts

```diff
@@ -94,6 +99,23 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
     };
   }
 
+  private async createServer(config: HttpConfig) {
+    const { port } = config;
+
+    if (!port || this.secondaryServers.has(port)) {
+      throw new Error(`port ${port} is already in use`);
+    }
+
+    const baseConfig = await this.config$.pipe(first()).toPromise();
+    const cfg = { ...baseConfig, ...config };
+    const log = this.logger.get('http', `server:${port}`);
+
+    const httpServer = new HttpServer(log);
+    const httpSetup = await httpServer.setup(cfg);
```

Comment:
when you are going to start secondary servers?

Reply:
Is there something a plugin would need to/could do to an HttpServer after creating it prior to starting it?  My idea was to let the plugin handle calling the start method in case it had other setup code to do.

If there's nothing that can be done after creating it before starting it, I can add start in here as well.

Reply:
the default schema was not working very well.  it was rejecting the `cors` with `[cors]: expected value of type [boolean] but got [Object]` (even though the cors value was coming from the config service; i wasn't defining it at all), and if I overwrote that value it then started saying ` [publicDir]: definition for this key is missing`.

I'm working on generating my own partial config for this.

Reply:
Yay this is resolved via #37118 -- you can now validate individual keys on an object schema, so this now loops through the keys passed in to verify they validate appropriately